### PR TITLE
fix: runix dev-shell

### LINKED ISSUE #29: `rust-env` flox environment is broken
If you run `flox develop -A rust-env` on a fresh clone using an `aarch64-darwin` machine you see the following error:
```
$ flox develop -A rust-env
warning: not writing modified lock file of flake 'git+file:///Users/zmitchell/code/floxdev/runix/wrap-uri-parser':
• Updated input 'flox-floxpkgs/nixpkgs/nixpkgs':
    'github:flox/nixpkgs/60c1d71f2ba4c80178ec84523c2ca0801522e0a6' (2023-03-19)
  → 'github:flox/nixpkgs/f742512ccc99e6ac457d6d4ddce78f283c4bbfde' (2023-06-25)
error:
       … while calling the 'derivationStrict' builtin

         at //builtin/derivation.nix:9:12: (source not available)

       … while evaluating derivation 'nix-shell'
         whose name attribute is located at /nix/store/3wla32r0vf8370p5y1zw1hk06914myvq-source/pkgs/stdenv/generic/make-derivation.nix:303:7

       … while evaluating attribute 'RUSTFMT' of derivation 'nix-shell'

         at /nix/store/f7c97habi7s8sxd7qgjd9bgaardd3vxi-source/shells/rust-env/default.nix:18:3:

           17|   RUST_SRC_PATH = rust.packages.stable.rustPlatform.rustLibSrc;
           18|   RUSTFMT = "${self.checks.pre-commit-check.passthru.rustfmt}/bin/rustfmt";
             |   ^
           19|   packages =

       error: function 'anonymous lambda' called with unexpected argument 'asNightly'

       at /nix/store/f7c97habi7s8sxd7qgjd9bgaardd3vxi-source/pkgs/rustfmt/default.nix:1:1:

            1| {rustfmt}: rustfmt.override {asNightly = true;}
             | ^
            2|
```

### PR DESCRIPTION
**Pull request by ysndr**
Develop runix using `nix/flox develop .#rust-env`

Closes #29

### CODE REVIEW

**Comment by ysndr** [comment]
Uncovered another bug with arg ordering command that prevents us from using `flox nix develop -c ...`

**Comment by zmitchell** [comment]
Why are we not using `flox` in CI anymore?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -24,15 +24,17 @@ jobs:
       with:
         github-access-token: ${{ secrets.NIX_GIT_TOKEN }}
 
+    # - flox develop does not support non-interactive shells at the moment
+    # - flox commands have command order isssues, to be fixed here in runix
+    #   and perculate to the install-flox-action
```
Comment (zmitchell) [inline .github/workflows/ci.yml:29]:
```suggestion
    #   and percolate to the install-flox-action
```

What are the command order issues?

**Review by zmitchell — APPROVED** [review]
If you haven't documented the "command order bug" you found somewhere you should do that so we know what's going wrong.

**Comment by ysndr** [comment]
I think this got mostly covered with #35

**Pull request closed without merging**